### PR TITLE
Creating add building page

### LINKED ISSUE #240: Create page for adding a new building
References user story #167 

We need a page to be able to add new buildings to the database

### PR DESCRIPTION
**Pull request by AshlynAdamson**
Closes #240 

Created an add building page, it's pretty simple. Currently, can only access it via the url, but I'm planning to have a button on the building list page that navigates the user to this page. 

image

### CODE REVIEW

File: BEIMA.Client/src/pages/Building/AddBuildingPage.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import { IoArrowBack } from "react-icons/io5";
+import { Card, Button, Form, Row, Col } from 'react-bootstrap';
+import { useEffect, useState } from "react";
+import styles from './AddBuildingPage.module.css';
+import FormListWithErrorFeedback from '../../shared/FormList/FormListWithErrorFeedback.js';
+
+
+const AddBuildingPage = () => {
+  // this will be replaced with API call based on selected device type to get the fields
+  const mandatoryBuildingFields = {
```
Comment (josephmoore99) [inline BEIMA.Client/src/pages/Building/AddBuildingPage.js:11]: I don't think it's necessary to have these referred to as "mandatory" as these should be the same fields for every building object.

File: BEIMA.Client/src/App.js [hunk 2]
```diff
@@ -31,6 +32,7 @@ const App = () => {
           <Route path="devices/:id" element={<DevicePage/>}/>
           <Route path="deviceTypes" element={<DeviceTypesPage/>}/>  
           <Route path="deviceTypes/:typeId" element={<DeviceTypePage/>}/>
+          <Route path="buildings/addBuilding" element={<AddBuildingPage/>}/>
```
Comment (josephmoore99) [inline BEIMA.Client/src/App.js:35]: Should we add a link on the navbar for this page?

Comment (AshlynAdamson) [reply BEIMA.Client/src/App.js:35]: I'm going to add one for buildings list, and then from that page there will be an "Add Building" button that will lead to this page

Comment (AshlynAdamson) [reply BEIMA.Client/src/App.js:35]: But we can add it to the nav bar, I just don't want to add too much, and the buildings will probably be something that isn't modified a whole lot, just to get the current buildings added.

File: BEIMA.Client/src/pages/Building/AddBuildingPage.js (hunk 1 above)
Comment (AshlynAdamson) [reply BEIMA.Client/src/pages/Building/AddBuildingPage.js]: Updated the name to allBuildingFields

**Review by josephmoore99 — APPROVED** [review]

**Review by roguedeleter — APPROVED** [review]

**Pull request merged**